src/domain/usecases/metadata/GetSharingSummaryUseCase.ts

```diff
@@ -12,10 +12,10 @@ export class GetSharingSummaryUseCase {
     public execute(update: SharingUpdate, payload: MetadataPayload): FutureData<SharingSummary> {
         const { baseElements, excludedDependencies } = update;
 
-        return this.metadataRepository.getMetadataWithChildren(baseElements).flatMap(payloads => {
-            const payloadsSummary = this.getPayloadsSummary(payloads, payload, excludedDependencies);
+        return this.metadataRepository.getMetadataWithChildren(baseElements).flatMap(selectedMetadata => {
```

Comment:
In general, try to improve the names of the variables and functions to facilitate the understanding of the code:
-  `selectedMetadata` represents the current metadata and `payload` represents the metadata with the new sharing settings. 
- In `getMetadataWithDifferentSharing` function you are mapping from payload to parent-children structure, not getting metadata with different sharing settings
- In `cleanMetadataSharing` function you are filtering the metadata that is going to have different sharing settings by comparing children-parent (so in this step you're actually getting metadata with different sharing settings to show the warning)